source/frontend/src/features/mapSideBar/lease/tabs/consultations/detail/ConsultationListView.tsx

```diff
@@ -0,0 +1,222 @@
+
+export interface IConsultationListViewProps {
+  loading: boolean;
+  consultations: ApiGen_Concepts_ConsultationLease[];
+  onAdd: () => void;
+  onEdit: (consultationId: number) => void;
+  onDelete: (consultationId: number) => void;
+}
+
+interface GroupedConsultations {
+  consultationTypeCode: string;
+  consultationTypeDescription: string;
+  consultations: ApiGen_Concepts_ConsultationLease[];
+  hasItems: boolean;
+}
+
+export const ConsultationListView: React.FunctionComponent<IConsultationListViewProps> = ({
+  loading,
+  consultations,
+  onAdd,
+  onEdit,
+  onDelete,
+}) => {
+  const keycloak = useKeycloakWrapper();
+  const history = useHistory();
+  const match = useRouteMatch();
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const { getOptionsByType } = useLookupCodeHelpers();
+  const consultationTypeCodes = getOptionsByType(API.CONSULTATION_TYPES);
+
+  const groupedConsultations = useMemo(() => {
+    const grouped: GroupedConsultations[] = [];
+    consultationTypeCodes.forEach(ct =>
+      grouped.push({
+        consultationTypeCode: ct.value.toString(),
+        consultationTypeDescription: ct.label,
+        consultations: consultations.filter(c => c.consultationTypeCode.id === ct.value),
+        hasItems: consultations.filter(c => c.consultationTypeCode.id === ct.value).length === 0,
+      }),
+    );
+    return grouped;
+  }, [consultationTypeCodes, consultations]);
+
+  if (loading) {
+    return <LoadingBackdrop show={loading} parentScreen={true} />;
+  }
+
+  return (
+    <StyledSummarySection>
+      <Section
+        header={
+          <SectionListHeader
+            claims={[Claims.LEASE_EDIT]}
+            title="Consultations"
+            addButtonText="Add Consultation"
+            addButtonIcon={<FaPlus size="2rem" />}
+            onAdd={onAdd}
+          />
+        }
+      >
+        {groupedConsultations.map((group, index) => (
+          <Section
+            key={index}
+            header={
+              <Row>
+                <Col xs="auto">
+                  <span className="pl-2">{group.consultationTypeDescription}</span>
+                </Col>
+                <Col className="px-0">
+                  {group.consultations.length > 0 && (
+                    <StyledIconWrapper>{group.consultations.length}</StyledIconWrapper>
+                  )}
+                </Col>
+              </Row>
+            }
+            noPadding
+            isCollapsable={!group.hasItems}
+            initiallyExpanded={group.hasItems}
+          >
+            {group.consultations.map((consultation, index) => (
+              <StyledBorder key={`consultation-section-${index}`}>
+                <Section
+                  header={
+                    <div>
+                      <Row>
+                        <Col>
+                          {group.consultationTypeCode === 'OTHER'
+                            ? `${consultation.otherDescription} Consultation`
```

Comment:
is null/undefined checking required here? aka ?? ''

Reply:
Not needed, the view model uses empty strings if the value coming from the wire is empty

Reply:
This is what I'm seeing locally:

![image](https://github.com/user-attachments/assets/a96a8451-c6de-47cc-8105-95bd5717ded2)